util/cache/cache.go

```diff
@@ -33,20 +33,31 @@ func NewCache(client CacheClient) *Cache {
 	return &Cache{client}
 }
 
-func buildRedisClient(redisAddress, password, username string, redisDB, maxRetries int, tlsConfig *tls.Config) *redis.Client {
-	opts := &redis.Options{
-		Addr:       redisAddress,
-		Password:   password,
-		DB:         redisDB,
-		MaxRetries: maxRetries,
-		TLSConfig:  tlsConfig,
-		Username:   username,
+func buildRedisClient(redisAddress, password, username string, redisDB, maxRetries int, tlsConfig *tls.Config, redisClusterMode bool) *redis.Client {
+	var client *redis.Client
+	if redisClusterMode {
+		opts := &redis.ClusterOptions{
+			Addrs:       []string{redisAddress},
```

Comment:
Need to be compatible with multiple addresses, use the form of split to return the address

Reply:
updated